src/main/java/ch/njol/skript/bukkitutil/PlayerUtils.java

```diff
@@ -38,71 +35,42 @@
- * 
- * @author Peter Güttinger
+ * Contains utility methods related to players
  */
 public abstract class PlayerUtils {
-	private PlayerUtils() {}
-	
-	final static Set<Player> inviUpdate = new HashSet<>();
-	
-	public static void updateInventory(final @Nullable Player p) {
-		if (p != null)
-			inviUpdate.add(p);
+
+	private static final Set<Player> inventoryUpdateList = Collections.synchronizedSet(new HashSet<>());
+
+	/**
+	 * Updates the clients inventory within a tick, using {@link Player#updateInventory()}.
+	 * Recommended over directly calling the update method,
+	 * as multiple calls to this method within a short timespan will not send multiple updates to the client.
+	 */
+	public static void updateInventory(@Nullable Player player) {
+		if (player != null)
+			inventoryUpdateList.add(player);
 	}
-	
-	// created when first used
-	final static Task task = new Task(Skript.getInstance(), 1, 1) {
-		@Override
-		public void run() {
-			try {
-				for (final Player p : inviUpdate)
+
+	static {
+		new Task(Skript.getInstance(), 1, 1) {
+			@Override
+			public void run() {
+				for (Player p : inventoryUpdateList)
 					p.updateInventory();
-			} catch (final NullPointerException e) { // can happen on older CraftBukkit (Tekkit) builds
-				if (Skript.debug())
-					e.printStackTrace();
+
+				inventoryUpdateList.clear();
 			}
-			inviUpdate.clear();
-		}
-	};
-	
-	private final static boolean hasCollecionGetOnlinePlayers = Skript.methodExists(Bukkit.class, "getOnlinePlayers", new Class[0], Collection.class);
-	@Nullable
-	private static Method getOnlinePlayers = null;
-	
-	@SuppressWarnings({"null", "unchecked"})
+		};
+	}
+
+	/**
+	 * @deprecated use {@link Bukkit#getOnlinePlayers()} instead
+	 */
+	@Deprecated
 	public static Collection<? extends Player> getOnlinePlayers() {
-		if (hasCollecionGetOnlinePlayers) {
-			return ImmutableList.copyOf(Bukkit.getOnlinePlayers());
-		} else {
-			if (getOnlinePlayers == null) {
-				try {
-					getOnlinePlayers = Bukkit.class.getDeclaredMethod("getOnlinePlayers");
-				} catch (final NoSuchMethodException e) {
-					Skript.outdatedError(e);
-				} catch (final SecurityException e) {
-					Skript.exception(e);
-				}
-			}
-			try {
-				final Object o = getOnlinePlayers.invoke(null);
-				if (o instanceof Collection<?>)
-					return ImmutableList.copyOf((Collection<? extends Player>) o);
-				else
-					return Arrays.asList(((Player[]) o).clone());
-			} catch (final IllegalAccessException e) {
-				Skript.outdatedError(e);
-			} catch (final IllegalArgumentException e) {
-				Skript.outdatedError(e);
-			} catch (final InvocationTargetException e) {
-				Skript.exception(e);
-			}
-			return Collections.emptyList();
-		}
+		return ImmutableList.copyOf(Bukkit.getOnlinePlayers());
 	}
-	
-	
+
 	public static boolean canEat(Player p, Material food) {
 		GameMode gm = p.getGameMode();
 		if (gm == GameMode.CREATIVE || gm == GameMode.SPECTATOR)
```

Comment:
I think you can now eat some foods in creative that give effects (magic stew, golden apple) but I'm not certain.
This might be outside the scope of this PR, though.

Reply:
I would agree with you on that last part. Besides, this method is currently unused.